qmi/instruments/newport/smc_100cc.py

```diff
@@ -30,3 +44,542 @@ def __init__(self,
+
+        self.controller_address = controller_address
+        _logger.info(
+            "Setting the max. output voltage of the driver to the motor of instrument [%s] to [%f]",
+            self._name, driver_voltage
+        )
+        self._scpi_protocol.write(self._build_command(
+            "DV", driver_voltage))
+        self._check_error()
+
+    @rpc_method
+    def get_low_pass_filter_cutoff_frequency(self, controller_address: Optional[int] = None) -> float:
+        """
+        Get the low pass filter cut-off frequency Kd.
+
+        Parameters:
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised value of the controller address.
+
+        Returns:
+            The low pass filter cut-off frequency, Kd, in Hertz.
+        """
+        _logger.info(
+            "Getting encoder increment value of instrument [%s]", self._name)
+        # instrument must be in configuration state to get the encoder increment value.
+        self.reset(controller_address)
+        self.enter_configuration_state(controller_address)
+        res = self._scpi_protocol.ask(
+            self._build_command("FD?"))
+        sleep(self.COMMAND_EXEC_TIME)
+        self._check_error()
+        self.exit_configuration_state(controller_address)
+        return float(res[3:])
+
+    @rpc_method
+    def set_low_pass_filter_cutoff_frequency(
+            self, frequency: float, persist: bool = False, controller_address: Optional[int] = None) -> None:
+        """
+        Set the low pass filter cut-off frequency Kd.
+
+        Parameters:
+            frequency:          Cutoff frequency Kd in Hertz.
+            persist:            Flag to indicate if the frequency cutoff should be persisted to the controller's memory,
+                                so it is still available after powering down the controller. When not persisted, the
+                                frequency cutoff is the one stored in the controller's memory.
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised frequency of the controller address.
+        """
+        if 2000 <= frequency or frequency <= 1E-6:
+            raise QMI_InstrumentException(
+                f"Provided value {frequency} not in valid range 1E-6 > frequency > 2000.")
+
+        _logger.info(
+            "Setting encoder increment frequency of instrument [%s] to [%s]", self._name, frequency)
+        self.controller_address = controller_address
+        # instrument must be in configuration state to persist the set velocity.
+        if persist:
+            self.reset(controller_address)
+            self.enter_configuration_state(controller_address)
+
+        self._scpi_protocol.write(self._build_command(
```

Comment:
Then I think what is needed is a check on the current state and then to switch to DISABLE if possible. I also noticed that the already implemented `enter_configuration_state` would benefit from such a check: If the state is NOT_REFERENCED (or already in CONFIGURATION), only then call `reset`. Otherwise we would be calling it unnecessarily.

Reply:
Although now trying to implement this I realize that adding these checks add complexity - and also add the need of more unit-tests per these commands. As we need to test then also for "if it was in READY state then did it set it to DISABLE to work?" kind of logic. I am not sure if it is worth it or should we just let it return the error in `_check_error` to user if the state was nor correct?

Reply:
"final" solution to this, to reduce complexity, was that I made the `enter/exit_configuration_state` calls INTERNAL functions and the RPC functions I replaced with `get/set_configuration_state`. I made a similar solution also for the `MM` command.